### PR TITLE
[Feat]: Continuously running carousel (ticker, marquee, ..)

### PR DESCRIPTION
**Pull request by davidjerleke**
- Closes: #114.

### CODE REVIEW

**Comment by yunusga** [comment]
HI @davidjerleke I use this effect from `v6.2.0` in `v7+` i found bug for vertical carousel with `loop`, loop work only if manualy drag the carousel/

I create this solution https://codesandbox.io/p/sandbox/embla-carousel-y-axis-vanilla-marquee-fvflyk just need animate scroll, for any direction

```
engine.scrollTo.distance(speed)
```

**Comment by davidjerleke** [comment]
@yunusga this PR isn’t finished. It doesn’t make sense to test it at this stage. I have a little left to do before this is ready for test.

File: packages/embla-carousel-auto-scroll/src/components/AutoScroll.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+      if (!options.stopOnInteraction) {
+        eventStore.add(root, 'mouseleave', () => {
+          resume = true
+          startScroll()
+        })
+      }
+    }
+
+    if (options.stopOnFocusIn) {
+      eventStore.add(root, 'focusin', () => {
```
Comment (davidjerleke) [inline packages/embla-carousel-auto-scroll/src/components/AutoScroll.ts:85]: Change `root` to `container`? Should also be done for the Autoplay plugin:

https://github.com/davidjerleke/embla-carousel/blob/9b2f60b293d4b45d2ede646ff34ce41829c4c9f8/packages/embla-carousel-autoplay/src/components/Autoplay.ts#L82-L86

File: packages/embla-carousel-auto-scroll/src/components/AutoScroll.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        stopScroll()
+        emblaApi.scrollTo(emblaApi.selectedScrollSnap(), true)
+      })
+
+      if (!options.stopOnInteraction) {
+        eventStore.add(root, 'focusout', startScroll)
```
Comment (davidjerleke) [inline packages/embla-carousel-auto-scroll/src/components/AutoScroll.ts:91]: Change `root` to `container`? Should also be done for the Autoplay plugin:

https://github.com/davidjerleke/embla-carousel/blob/9b2f60b293d4b45d2ede646ff34ce41829c4c9f8/packages/embla-carousel-autoplay/src/components/Autoplay.ts#L82-L86

**Review by davidjerleke — COMMENTED** [review]
We should change the element that has a focus listener from `root` to `container`, in order to prevent previous/next buttons and other focusable things outside the container (but inside the root) from stopping the carousel.

**Comment by decapination-labs** [comment]
any docs on how to use it ? thx

**Comment by davidjerleke** [comment]
@decapination-labs [docs here](https://www.embla-carousel.com/plugins/auto-scroll/), and [CodeSandbox here](https://github.com/davidjerleke/embla-carousel/issues/114#issuecomment-1891071784), but:

> [!WARNING]  
> You should [read this](https://github.com/davidjerleke/embla-carousel/issues/114#issuecomment-1911757602) first 😢.

**Comment by decapination-labs** [comment]
yes i am waiting for the updates !! something to point we dont have onMouseHover? and onTouchIn ?

**Comment by davidjerleke** [comment]
> something to point we dont have onMouseHover? and onTouchIn ?

@decapination-labs could you elaborate? I don't understand what you mean?

**Comment by decapination-labs** [comment]
if i hover my mouse on the carousel item ( without clicking ) , the autoplay stops !! 

the same for mobile on touch

**Comment by davidjerleke** [comment]
Have you checked the following options?

- [https://www.embla-carousel.com/plugins/auto-scroll/#stoponinteraction](stoponinteraction)
- [https://www.embla-carousel.com/plugins/auto-scroll/#stoponmouseenter](stoponmouseenter)

You should be able to configure it.

**Comment by decapination-labs** [comment]
oh sorry i think i messed up with the options delay and duration before u release the upates to have a close ux haha and looks like the stoponmouseenter works fine . plz let me know when u release because need that feature even though i have to tweak it a bit because i want continious smooth sliding and dragfree without reInit maybe or something other api calls which affecting the (continue smooth scroll) after i release the mouse

**Comment by davidjerleke** [comment]
@decapination-labs please take some time to read the API to better understand how it works [here](https://www.embla-carousel.com/plugins/auto-scroll/).

**Comment by decapination-labs** [comment]
i am facing an issue , i am creating 2 parallel carousel each sliding in different direction ! (nextjs + shadcnui)

https://codesandbox.io/p/sandbox/embla-carousel-generator-react-forked-k5h2lm?file=%2Fsrc%2Fjs%2Findex.tsx%3A15%2C13

this looks ok for sliding rtl but if i change that direction to ltr then the carousel starts to behave weird !! can u help plz

**Comment by davidjerleke** [comment]
@decapination-labs you're using the `Autoplay` plugin and not `AutoScroll`? This PR is for the AutoScroll plugin only.

**Comment by davidjerleke** [comment]
@decapination-labs have you read the docs about the [`direction`](https://www.embla-carousel.com/api/options/#direction) options? It says:

> [!NOTE]  
> **Note:** When using `rtl`, the content direction also has to be set to RTL, either by using the [HTML dir attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir) or the [CSS direction](https://developer.mozilla.org/en-US/docs/Web/CSS/direction) property.

**Comment by decapination-labs** [comment]
ohhhh i missed that note .. thank you again and i am waiting the autoscroll publish !!

**Comment by decapination-labs** [comment]
> @decapination-labs [docs here](https://www.embla-carousel.com/plugins/auto-scroll/), and [React CodeSandbox here](https://github.com/davidjerleke/embla-carousel/issues/114#issuecomment-1891071784), but:
> 
> Warning
> 
> You should [read this](https://github.com/davidjerleke/embla-carousel/issues/114#issuecomment-1911757602) first 😢.

is it published or not yet ?

**Comment by davidjerleke** [comment]
@decapination-labs the person who had the name on npm unpublished their package yesterday so that frees up the name. BUT, for reasons I don’t know, npm blocks any new release for that name for 24h. It hasn’t been 24h yet.

**Comment by decapination-labs** [comment]
ok cool . will be waiting the update . thanks

**Pull request merged**